samza-core/src/main/java/org/apache/samza/zk/ZkBarrierForVersionUpgrade.java

```diff
@@ -244,10 +239,8 @@ public void handleDataChange(String dataPath, Object data) {
     }
 
     @Override
-    public void handleDataDeleted(String dataPath) {
-      LOG.warn("barrier done node got deleted at " + dataPath);
-      if (notAValidEvent())
-        return;
+    public void doHandleDataDeleted(String path) {
+      LOG.warn("Data deleted in path: " + path);
```

Comment:
Minor: This message would sound alarming in logs. Possible to add more context here about what might be wrong?